### PR TITLE
[XPU]Migrate adamw XPU kernel into Phi

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR types

Others
### PR changes

OPs
### Describe

[XPU]Migrate adamw XPU kernel into Phi.

### CODE REVIEW

File: paddle/phi/kernels/xpu/adamw_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                       dev_ctx.template Alloc<T>(param_out),
+                       beta1_,
+                       beta2_,
+                       epsilon_,
+                       coeff,
+                       param.numel());
+    PADDLE_ENFORCE_EQ(
+        r,
+        xpu::SUCCESS,
+        errors::External("XPU kernel adamw occur error in adamw error code ",
```
Comment (ykkk2333) [inline paddle/phi/kernels/xpu/adamw_kernel.cc]: PADDLE_ENFORCE_XDNN_SUCCESS，可以使用这个检查

File: paddle/phi/kernels/xpu/adamw_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                     beta2_pow_out_p,
+                     beta2_pow.numel(),
+                     false,
+                     beta2_,
+                     0.0f);
+      PADDLE_ENFORCE_EQ(
+          r,
+          xpu::SUCCESS,
+          errors::External("XPU kernel scale occur error in adamw error code ",
+                           r,
```
Comment (ykkk2333) [inline paddle/phi/kernels/xpu/adamw_kernel.cc]: 同上

File: paddle/phi/kernels/xpu/adamw_kernel.cc (hunk 1 above)
Comment (Aurelius84) [reply paddle/phi/kernels/xpu/adamw_kernel.cc]: thanks, fixed

File: paddle/phi/kernels/xpu/adamw_kernel.cc (hunk 2 above)
Comment (Aurelius84) [reply paddle/phi/kernels/xpu/adamw_kernel.cc]: thanks, fixed

**Review by YuanRisheng — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]

**Pull request merged**